Homepage: Add hero section for random featured site

ryelle opened the pull request:
Fixes #135. This adds a random site to the header as a "hero". The site is chosen from the "sticky posts" to avoid creating a second Featured category (the current category "Featured" is used for the homepage grid). The same approach was used in the first redesign iteration, see https://github.com/WordPress/wporg-showcase-2022/pull/37.

The Hero section displays a heading & blurb for Showcase, then the featured site's meta: author, domain, and category (should this be tags?).

| Has all 3 meta fields | Missing "author" | 
|---|---|
| ![Screen Shot 2023-08-14 at 18 11 34](https://github.com/WordPress/wporg-showcase-2022/assets/541093/653cc073-c18e-4dc8-a98f-8e4f9809d028) | ![Screen Shot 2023-08-14 at 18 11 18](https://github.com/WordPress/wporg-showcase-2022/assets/541093/88b67e79-6d42-4c25-acb0-c5cdf8ae50b4) |

The responsive styles are a work in progress, this will depend on how the spacing conversation goes I think. Currently the hero breaks into one column at 1179px, because that's when the image starts to look too squished. At 480px, the spacing and border are definitely too big, but that also depends on the outcome of [the spacing issue](https://github.com/WordPress/wporg-main-2022/issues/296).

| 1080px | 1000px | 480px |
|---|---|---|
| ![Screen Shot 2023-08-14 at 18 29 51](https://github.com/WordPress/wporg-showcase-2022/assets/541093/969c32d8-f659-46f8-a7ae-0502a1538267) | ![Screen Shot 2023-08-14 at 18 27 09](https://github.com/WordPress/wporg-showcase-2022/assets/541093/8d3b1419-c6a3-43be-94f0-58150061cdd8) | ![Screen Shot 2023-08-14 at 18 27 26](https://github.com/WordPress/wporg-showcase-2022/assets/541093/37e73c90-249a-480f-a2a9-ccca75f5d3fb) |

**To test**

- Check out branch and build
- Add some sticky posts
- View the homepage
- It should show one of the sites you made sticky
- The site info should show, and the desktop-sized image should be a link to the site detail page
- Reload (maybe a few times if only a few sticky), the featured site should change randomly

Due to the override we're using to fix [this issue](https://github.com/WordPress/gutenberg/issues/41184), if you don't have any sticky posts, it will show any random site.

adamwoodnz commented:
Comparing against the design at monitor size (2560w) the dots look a bit small and the title a little high

### Screenshots

| Design | Browser |
|-|-|
| ![(Monitor Extra Large) Showcase - Landing](https://github.com/WordPress/wporg-showcase-2022/assets/1017872/bc5c36d5-eb66-4f09-8946-752a77732861) | ![localhost_8888_(Monitor)](https://github.com/WordPress/wporg-showcase-2022/assets/1017872/ea907236-d513-46b1-b691-da5f9bcf18df) |

adamwoodnz commented:
And from 1180px > 1760px I don't have any left padding

<img src="https://github.com/WordPress/wporg-showcase-2022/assets/1017872/ceaa256f-2ebf-447e-becb-e5df8415db2d" width="500">

adamwoodnz commented on source/wp-content/themes/wporg-showcase-2022/patterns/_site-hero.php:
```diff
@@ -0,0 +1,55 @@
+					<!-- /wp:spacer -->
+
+					<!-- wp:heading {"fontSize":"heading-2"} -->
+					<h1 class="wp-block-heading has-heading-2-font-size"><?php esc_attr_e( 'Showcase', 'wporg' ); ?></h1>
+					<!-- /wp:heading -->
+
+					<!-- wp:paragraph {style":{"spacing":{"margin":{"top":"var:preset|spacing|10"}}}} -->
+					<p style="margin-top:var(--wp--preset--spacing--10)"><?php esc_attr_e( 'Star-studded sites built with WordPress', 'wporg' ); ?></p>
```
These translations aren't attributes, is `esc_attr_e` appropriate?

ryelle replied:
Nope, I must've copied this and forgot to change it. thanks!

adamwoodnz commented on source/wp-content/themes/wporg-showcase-2022/patterns/_site-hero.php:
```diff
@@ -0,0 +1,55 @@
+
+			<!-- wp:column {"verticalAlignment":"bottom"} -->
+			<div class="wp-block-column is-vertically-aligned-bottom">
+				<!-- wp:cover {"url":"http://localhost:8888/wp-content/themes/wporg-showcase-2022/images/dots-dark.svg","isRepeated":true,"dimRatio":0,"focalPoint":{"x":0.5,"y":0},"minHeight":440,"contentPosition":"bottom center","style":{"spacing":{"padding":{"top":"70px","right":"70px","bottom":"0","left":"70px"}}},"layout":{"type":"constrained"}} -->
+				<div class="wp-block-cover is-repeated has-custom-content-position is-position-bottom-center" style="padding-top:70px;padding-right:70px;padding-bottom:0;padding-left:70px;min-height:440px">
+					<span aria-hidden="true" class="wp-block-cover__background has-background-dim-0 has-background-dim"></span>
+					<div role="img" class="wp-block-cover__image-background is-repeated" style="background-position:50% 0%;background-image:url(http://localhost:8888/wp-content/themes/wporg-showcase-2022/images/dots-dark.svg)"></div>
```
Assume you'll upload the image to staging and replace these `localhost` urls

ryelle replied:
🤦🏻

ryelle replied:
Updated in https://github.com/WordPress/wporg-showcase-2022/pull/145/commits/1798bb1ce3a1ffea4d9bf1a66bb61326318f11ba

adamwoodnz reviewed (commented):
Looking good, a few comments inline

jasmussen commented:
Nice PR, lots to like here. I'd appreciate an eye by @fcoveram for a sanity check on the metrics, especially the max-width. here, 1920 from the mockups:

<img width="2174" alt="Screenshot 2023-08-15 at 10 27 54" src="https://github.com/WordPress/wporg-showcase-2022/assets/1204802/d7239971-dfbf-490b-bbea-2fcccdf8f8d0">

In this case the dots actually scale a bit up compared to the 1320 view:

<img width="1547" alt="Screenshot 2023-08-15 at 10 28 54" src="https://github.com/WordPress/wporg-showcase-2022/assets/1204802/c9b54a8d-2e76-4ef9-a1b7-702f7235e2a3">

At a glance, hard to test, the stroke in the mobile view looks a bit thick. It may be a trick of my eyes, and might not be something to fix here. But an option is to look at vw scaling strokes, like so: https://codepen.io/joen/pen/GRwXPop

ryelle commented:
I fixed the code issues, and have pulled out the following issues from above (paraphrased). A lot is waiting on the spacing conversation.

> Showcase title is lower on large screens

This will depend on the new spacing values, I believe I grabbed the midsized-screen value to work with for now. On the meeting last week we decided that would work until the new spacing values were finalized. https://github.com/WordPress/wporg-main-2022/issues/296

> Dots are larger on large screens

Ah, I missed this on the design. I tried a version with clamp, it's mostly a guess to see if that would work, but it's roughly there. Maybe once this is pushed to staging a designer can take a look & use inspector to figure out the right values? Also to note, the spacing above the screenshot is a little awkward combined with the scaling dots, but when the spacing values are worked out this should scale too.

> Meta fields should have more padding in general and even more padding on large screens

Currently this uses the same block as the single site meta list, so it was just inheriting the default spacing. I've updated the spacing when used without a label (like in the hero), but haven't address the scaling based on screen size (again, waiting on the spacing convo).

4. Stroke should scale with viewport size

Okay, I've grabbed the `clamp` from that codepen and added it.

Latest screenshots:

| Width | Screenshot |
|---|---|
| 1920 | ![ss-1920](https://github.com/WordPress/wporg-showcase-2022/assets/541093/36a95c18-fdf6-4a6d-8334-8e8eb5bef971) |
| 1320 | ![ss-1320](https://github.com/WordPress/wporg-showcase-2022/assets/541093/be9654a7-ac24-4e3f-9ec9-726bdf56fcef) |
| 960 | ![ss-960](https://github.com/WordPress/wporg-showcase-2022/assets/541093/1b70953d-15cd-45ea-96fd-c779a529e4fb) |
| 600 | ![ss-600](https://github.com/WordPress/wporg-showcase-2022/assets/541093/650dd0e3-5080-4e16-a215-464c855be756) |
| 480 | ![ss-480](https://github.com/WordPress/wporg-showcase-2022/assets/541093/e50a14ab-0320-4f80-a4c4-c5054b611b7a) |

StevenDufresne commented:
Nice! This is looking good. A few descrepancies I've noticed:

- [ ] Link color
- [ ] Link decoration
- [ ] On Mobile the "Showcase" title should be hidden and the subnav visible (maybe this comes later).

## Thoughts

Apologies for the late feedback, seeing this live has brought this out, but I was confused as to what the details in the "box" were especially because i don't know who/what "Conde Nast" is, haven't seen the meta detail box anywhere yet and the leading text, "Star-studded sites built with WordPress", seems to introduce the meta details but doesn't. Based on how it's written I would expect to see a list of sites.


@marko-srb Did we consider adding labels next to fields for the homepage? Too verbose?

| Before | After |
| --- | --- |
| <img width="338" alt="Screenshot 2023-08-16 at 11 55 49 AM" src="https://github.com/WordPress/wporg-showcase-2022/assets/1657336/7873a50c-8df8-4fa2-b6ef-3a8e5b5909e6"> | <img width="332" alt="Screenshot 2023-08-16 at 11 55 39 AM" src="https://github.com/WordPress/wporg-showcase-2022/assets/1657336/a13c0f22-15d0-49ec-b251-b5c0d49397d5"> | 

Also, would we consider dropping the meta details on mobile? Would that be a bad idea? The image is what pops, but instead I get a list of information that seems secondary and I am very unlikely to click on.

| What I see | What I want to see |
| --- | --- |
| <img width="338" alt="Screenshot 2023-08-16 at 11 55 49 AM" src="https://github.com/WordPress/wporg-showcase-2022/assets/1657336/f0c671c3-43b2-4439-a858-74c151e46c61"> | <img width="394" alt="Screenshot 2023-08-16 at 12 09 14 PM" src="https://github.com/WordPress/wporg-showcase-2022/assets/1657336/1d3da98a-45c7-4674-b1aa-4813943f822d"> |

jasmussen commented:
At a glance, looks good! I'd appreciate a sanity check by @fcoveram 

Probably nothing to block this PR, we can always follow up.

ryelle commented:
@StevenDufresne @marko-srb For what it's worth, this is using the same block as the site details meta box, it just hides the labels. I could just untoggle that:

<img width="438" alt="Screenshot 2023-08-16 at 10 16 38 AM" src="https://github.com/WordPress/wporg-showcase-2022/assets/541093/e19a010d-a83d-4b0e-9d78-7c00734071d0">

marko-srb commented:
Just a thing,

There's no 'author' as metadata on the landing. The first row is actually just the name of the site in full. 
So that field should be always present.

We made it through several pushbacks on the home having 3 metadata points:

1. Name (must have)
2. UIR (must have)
3. Tags (must have)

Sorry if this wasn't clear enough...

marko-srb commented:
For the labels thing. 

I'd say we go without them on Home page. If we consider the name, url and tags, I am not sure we really need labels? Maybe it gets confusing sometimes, but I guess the name of the site is first what you see, and in most cases you see the same name on the site screenshot on the right (or initials) — at least that's the case for 6 sites I narrowed in showcase hero section for landing...

If we go with a label though, it should say 'site name' not 'author', as we would avoid having less than those 3 data there on hero section... [ref is previous comment]

marko-srb commented:
> 
@StevenDufresne @marko-srb For what it's worth, this is using the same block as the site details meta box, it just hides the labels. I could just untoggle that:

Issue with that is, it can become a bit problematic in layout area, due to hero section attempt to have everything left aligned. But if you pursue this, please do test it, it may look better than I think and it is a quick test...

marko-srb commented:
> @StevenDufresne @marko-srb For what it's worth, this is using the same block as the site details meta box, it just hides the labels. I could just untoggle that:
> 
> <img alt="Screenshot 2023-08-16 at 10 16 38 AM" width="438" src="https://user-images.githubusercontent.com/541093/261036703-e19a010d-a83d-4b0e-9d78-7c00734071d0.png">


Issue with that is, it can become a bit problematic in layout area, due to hero section attempt to have everything left aligned. But if you pursue this, please do test it, it may look better than I think and it is a quick test...

ryelle commented:
> There's no 'author' as metadata on the landing. The first row is actually just the name of the site in full.

Ah, okay. I made an incorrect assumption. In the future, could you annotate that sort of thing in the mockups (either in figma or when posting in github)? That would help.

> If we consider the name, url and tags, I am not sure we really need labels?

I think that's fair, we can go without labels for now and if it seems unclear when merged, update it then 👍🏻

jasmussen commented:
Nice, thanks for the update. 

In general, links in the metadata boxes are not underlined, and on the homepage they have a different link color:

<img width="942" alt="Screenshot 2023-08-21 at 14 06 22" src="https://github.com/WordPress/wporg-showcase-2022/assets/1204802/684e121c-478d-4620-996f-96d3aeab3faf">

We can keep the external link icon as you have it for now, though there's a separate conversation on whether this should be the ↗ unicode icon instead, standardizing this on external links. CC: @fcoveram

fcoveram commented:
### External link icon

Agree with @jasmussen on replacing the external icon with the unicode symbol.

### Thumbnail on mobile

I'm not sure what is the last version of hero section on mobile, but worth mention that in design the image is set to be full-width and reach the main container. See the red guidelines in the image below. Sorry the low contrast of the marks.

<img width="500" alt="Hero section of Showcase design with guidelines" src="https://github.com/WordPress/wporg-showcase-2022/assets/895819/104e3c33-96eb-4564-8659-1900bef4d52b">

jasmussen commented:
Go for it!
